monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,324 @@
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(70) # build a buffer so that the radius is 70m for the track
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
+            flight_points_with_altitude = []
+            x, y = buffered_path.exterior.coords.xy
+            for coord in range(0,len(x)):
+                flight_points_with_altitude.append(FlightPoint(lat = x[coord],lng = y[coord],alt= altitude))
+
+            # Build a list of points so that they can be fed to the sim and outputted. 
+            self.flight_points.append(flight_points_with_altitude)
+
+
+class TrackWriter():
+
+    """
+        Write the tracks created by AdjacentCircularFlightPathsGenerator into disk (in the outputs directory) as GeoJSON FeatureCollection 
+        Args:
+        flight_path_points: A set of flight path points generated by generate_flight_paths_points method in the AdjacentCircularFlightPathsGenerator class
+        bboxes: A set of bounding boxes generated by generate_query_bboxes method in the AdjacentCircularFlightPathsGenerator class
+        country_code: An ISO 3166-1 alpha-3 code for a country
+
+        Outputs: 
+        GeoJSON files for bboxes created in the `test_definitions/{country_code}` folder 
+        
+        
+    """
+
+    def __init__(self, path_points, bboxes, country_code='CHE') -> None:
+
+        self.flight_path_points = path_points
+        self.bboxes = bboxes
+        self.country_code = country_code        
+        self.output_directory = Path('test_definitions', self.country_code)
+        self.output_directory.mkdir(parents=True, exist_ok=True) # Create test_definition directory if it does not exist
+
+    def write_bboxes(self):
+
+        ''' This module writes the bboxes as a GeoJSON FeatureCollection '''
+        for buffered_bbox_details in self.bboxes:
+            
+            features = json.dumps({'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(buffered_bbox_details.shape)})
+            bbox_file_name = 'box_%s.geojson'% buffered_bbox_details.name
+            bbox_output_path = self.output_directory / bbox_file_name
+
+            with open(bbox_output_path,'w') as f:
+                f.write(features)
+                
+    def write_tracks(self):
+
+        ''' This module writes tracks as a GeoJSON FeatureCollection (of Point Feature) for use in other software '''       
+        
+        flight_point_lenghts = {}
```

Comment:
Is this variable or the following one used anywhere?

Reply:
No it was used earlier removed now.

Reply:
Confirmed that this is corrected now.